### PR TITLE
ISSUE-410 Adopt eclipse-temurin:18-jre-focal docker image

### PR DESCRIPTION
**Pull request by vttranlina**
https://github.com/linagora/tmail-backend/issues/410

### CODE REVIEW

**Comment by vttranlina** [comment]
WIP
for CI checking

File: pom.xml [hunk 1]
```diff
@@ -30,6 +30,22 @@
...
+                                <jvmFlag>-Dlogback.configurationFile=/root/conf/logback.xml</jvmFlag>
+                                <jvmFlag>-Dworking.directory=/root/</jvmFlag>
+                                <!-- Prevents Logjam (CVE-2015-4000) -->
+                                <jvmFlag>-Djdk.tls.ephemeralDHKeySize=2048</jvmFlag>
+                                <jvmFlag>--add-opens=java.base/java.lang=ALL-UNNAMED</jvmFlag>
+                                <jvmFlag>--add-opens=java.base/java.math=ALL-UNNAMED</jvmFlag>
+                                <jvmFlag>--add-opens=java.base/java.util=ALL-UNNAMED</jvmFlag>
+                                <jvmFlag>--add-opens=java.base/java.util.concurrent=ALL-UNNAMED</jvmFlag>
+                                <jvmFlag>--add-opens=java.base/java.net=ALL-UNNAMED</jvmFlag>
+                                <jvmFlag>--add-opens=java.base/java.text=ALL-UNNAMED</jvmFlag>
```
Comment (Arsnael) [inline pom.xml:45]: Just curious, why we need all those jvm flags?

Comment (vttranlina) [reply pom.xml:45]: As I research, We use maven compile on java 11, then we try to run it on java 18 (here is eclipse java 18-jre-forcal).
We got the error `java.lang.reflect.InaccessibleObjectException` when run deployment-test. 
Some topics on StackOverflow guide this solution.

Comment (vttranlina) [reply pom.xml:45]: Maybe we don't need them when we use maven compile on java 17. 
Let me make another pr

**Comment by chibenwa** [comment]
That would be good to plan some performance tests to validate this change (before / after).

**Comment by vttranlina** [comment]
> That would be good to plan some performance tests to validate this change (before / after).

Should we monitor Glowroot? (or just only Gatling report?)
If yes with Glowroot, Can you give me some focus indicates?

**Comment by chibenwa** [comment]
just only Gatling reports is OK

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
I'd like to have gatling reports for IMAP before merging...

**Comment by vttranlina** [comment]
# Performance test result  
- Local env
- docker-compose distributed
- Gatling result: [jre11-jre18_imap-jmap.zip](https://github.com/linagora/tmail-backend/files/8675040/jre11-jre18_imap-jmap.zip)
## IMAP
- ImapFullWithDurationSimulation
### Before
image

### After
image


## JMAP
- AllSimulation
### Before
image

### After 
image

**Comment by vttranlina** [comment]
Response time improvement

**Comment by vttranlina** [comment]
response time has been improvement

**Comment by chibenwa** [comment]
Can you please try with more requests ?

**Comment by Arsnael** [comment]
Wouldn't it be better to try to deploy that on preprod and test on it?

**Comment by vttranlina** [comment]
Round 2
[jre11-jre18_2.zip](https://github.com/linagora/tmail-backend/files/8676235/jre11-jre18_2.zip)
 
## Before
image

## After

image

**Comment by chibenwa** [comment]
(can we do yet 10x more?) ^^

Overwize yes very promissing!

**Review by chibenwa — APPROVED** [review]

**Comment by vttranlina** [comment]
> That's not preprod right?

Yes,

**Comment by vttranlina** [comment]
Round 3
- Env: Preprod
- org.apache.james.gatling.simulation.imap.PlatformValidationSimulation
[jre11-jre18_3.zip](https://github.com/linagora/tmail-backend/files/8678635/jre11-jre18_3.zip)

## Before
image

## After
image

**Comment by vttranlina** [comment]
Docker images for lab: https://hub.docker.com/repository/docker/tungtv202/tmail-backend-distributed

**Comment by chibenwa** [comment]
> Round 3

Be less shy with the count of users

Personnally I did runs at 15.000 users, injection time 40 mins, testing time 60 mins.

Please concider JVM warm up: code is first interpreted, then compiled and optimized (in two passes) using JIT (just in time compilation). The workload you test are far from impacting enough to even be remotely close from exercising GC changes, and generally a system under load is behaving differently from a system being not loaded (and here is is not loaded!).

**Comment by vttranlina** [comment]
> Personnally I did runs at 15.000 users, injection time 40 mins, testing time 60 mins.

I have a bit surprised, 
I tried 200 users per second and DURATION=20 minutes, 
and I wait for 2 hours, Gatling still not completed

**Comment by Arsnael** [comment]
preprod is supposed to be battle tested, don't go so little on it :)

**Comment by vttranlina** [comment]
Round 4
- preprod
- DURATION =1 hour , User 15000
[jre11-jre18_4.zip](https://github.com/linagora/tmail-backend/files/8685241/jre11-jre18_4.zip)

## Before
image

## After
image

**Comment by Arsnael** [comment]
Seems it looks better overall :)

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**